ringhold/src/intTest/java/org/opencadc/ringhold/InventoryValidatorTest.java

```diff
@@ -0,0 +1,323 @@
+ */
+public class InventoryValidatorTest {
+
+    private static final Logger log = Logger.getLogger(InventoryValidatorTest.class);
+
+    static {
+        Log4jInit.setLevel("org.opencadc.inventory", Level.INFO);
+        Log4jInit.setLevel("org.opencadc.inventory.db", Level.INFO);
+        Log4jInit.setLevel("ca.nrc.cadc.db", Level.INFO);
+        Log4jInit.setLevel("org.opencadc.ringhold", Level.INFO);
+    }
+
+    static String INVENTORY_SERVER = "RINGHOLD_TEST";
+    static String INVENTORY_DATABASE = "cadctest";
+    static String INVENTORY_SCHEMA = "inventory";
+    static String TMP_DIR = "build/tmp";
+    static String USER_HOME = System.getProperty("user.home");
+
+    static {
+        try {
+            File opt = FileUtil.getFileFromResource("intTest.properties", InventoryValidatorTest.class);
+            if (opt.exists()) {
+                Properties props = new Properties();
+                props.load(new FileReader(opt));
+
+                if (props.containsKey("inventoryServer")) {
+                    INVENTORY_SERVER = props.getProperty("inventoryServer").trim();
+                }
+                if (props.containsKey("inventoryDatabase")) {
+                    INVENTORY_DATABASE = props.getProperty("inventoryDatabase").trim();
+                }
+                if (props.containsKey("inventorySchema")) {
+                    INVENTORY_SCHEMA = props.getProperty("inventorySchema").trim();
+                }
+            }
+        } catch (MissingResourceException | FileNotFoundException noFileException) {
+            log.debug("No intTest.properties supplied.  Using defaults.");
+        } catch (IOException oops) {
+            throw new RuntimeException(oops.getMessage(), oops);
+        }
+
+        log.debug("intTest database config: " + INVENTORY_SERVER + " " + INVENTORY_DATABASE + " " + INVENTORY_SCHEMA);
+    }
+
+    final ArtifactDAO artifactDAO;
+    final DeletedStorageLocationEventDAO deletedStorageLocationEventDAO;
+    final Map<String, Object> daoConfig = new TreeMap<>();
+    final String jndiPath = "jdbc/InventoryEnvironment";
+
+    public InventoryValidatorTest() throws Exception {
+        artifactDAO = new ArtifactDAO();
+        deletedStorageLocationEventDAO = new DeletedStorageLocationEventDAO();
+
+        final DBConfig dbrc = new DBConfig();
+        final ConnectionConfig connectionConfig = dbrc.getConnectionConfig(INVENTORY_SERVER, INVENTORY_DATABASE);
+        DBUtil.createJNDIDataSource(jndiPath, connectionConfig);
+
+        daoConfig.put(SQLGenerator.class.getName(), SQLGenerator.class);
+        daoConfig.put("jndiDataSourceName", jndiPath);
+        daoConfig.put("database", INVENTORY_DATABASE);
+        daoConfig.put("schema", INVENTORY_SCHEMA);
+
+        artifactDAO.setConfig(daoConfig);
+        deletedStorageLocationEventDAO.setConfig(daoConfig);
+    }
+
+    @Before
+    public void setup() throws Exception {
+        writeConfig();
+        truncateTables();
+    }
+
```

Comment:
Final thing I didn't notice the other day. I have generally found that it is best to do cleanup to a known state @before and not @after. That way, the last test run leaves it's content behind and if a test is failing a dev can run that one test and examine the database state manually afterward. Leaving a little content from the last test behind is generally fine - other tests using that database have to cleanup before to be safe anyway.

Also, in the constructor, you should invoke the InitDatabase facility in the db lib. Kill your pg container and start a new one to verify that works.

Reply:
Updated the int test to use initDatabase, and that does work, just have to give the pg container a couple of secs to startup before running the intTest.